runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -103,121 +97,118 @@
 	@Inject
 	private AbyssOverlay abyssOverlay;
 
+	@Inject
+	private RunecraftOverlay runecraftOverlay;
+
 	@Inject
 	private RunecraftConfig config;
 
 	@Inject
 	private Notifier notifier;
 
 	@Provides
-	RunecraftConfig getConfig(ConfigManager configManager)
-	{
+	RunecraftConfig getConfig(ConfigManager configManager) {
 		return configManager.getConfig(RunecraftConfig.class);
 	}
 
 	@Override
-	protected void startUp() throws Exception
-	{
+	protected void startUp() throws Exception {
 		overlayManager.add(abyssOverlay);
 		abyssOverlay.updateConfig();
+		overlayManager.add(runecraftOverlay);
 	}
 
 	@Override
-	protected void shutDown() throws Exception
-	{
+	protected void shutDown() throws Exception {
 		overlayManager.remove(abyssOverlay);
 		abyssObjects.clear();
 		darkMage = null;
 		degradedPouchInInventory = false;
+		overlayManager.remove(runecraftOverlay);
 	}
 
 	@Subscribe
-	public void onConfigChanged(ConfigChanged event)
-	{
+	public void onConfigChanged(ConfigChanged event) {
 		abyssOverlay.updateConfig();
 	}
 
 	@Subscribe
-	public void onChatMessage(ChatMessage event)
-	{
-		if (event.getType() != ChatMessageType.GAMEMESSAGE)
-		{
+	public void onChatMessage(ChatMessage event) {
+		if (event.getType() != ChatMessageType.GAMEMESSAGE) {
 			return;
 		}
 
-		if (config.degradingNotification())
-		{
-			if (event.getMessage().contains(POUCH_DECAYED_MESSAGE))
-			{
+		if (config.degradingNotification()) {
+			if (event.getMessage().contains(POUCH_DECAYED_MESSAGE)) {
 				notifier.notify(POUCH_DECAYED_NOTIFICATION_MESSAGE);
 			}
 		}
 	}
 
 	@Subscribe
-	public void onMenuEntryAdded(MenuEntryAdded entry)
-	{
-		if (!wearingTiara)
-		{
+	public void onMenuEntryAdded(MenuEntryAdded entry) {
+		if (client.getGameState() != GameState.LOGGED_IN) {
+			return;
+		}
+
+		Widget loginScreenOne = client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN);
+		Widget loginScreenTwo = client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN_MESSAGE_OF_THE_DAY);
+
+		if (loginScreenOne != null || loginScreenTwo != null) {
 			return;
 		}
 
 		final String option = Text.removeTags(entry.getOption()).toLowerCase();
 		final String target = Text.removeTags(entry.getTarget()).toLowerCase();
-		final int id = entry.getIdentifier();
-		final int type = entry.getType();
 
-		if (target.contains("pouch") && !target.startsWith("rune"))
-		{
-			if (option.equals("deposit-all") && type == 57 && id == 2)
-			{
-				swap(client, "fill", option, target);
-				swap(client, "cancel", option, "", target);
-			}
-			else  if (option.equals("fill") && id != 9)
-			{
-				swap(client, "empty", option, target);
-			}
-		}
 
-		else if (target.contains("ring of dueling") && option.contains("remove"))
-		{
-			if (target.contains("7") || target.contains("5") || target.contains("3") || target.contains("1"))
-			{
-				swap(client, "castle wars", option, target);
+		Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
```

Comment:
the way I did this (with deposit-all/type/id) made sure it only worked in bank as well, this is a unneeded  change